### PR TITLE
Fingerprint request in export and backup

### PR DESCRIPTION
**Pull request by JDonadio**
Fixes https://github.com/bitpay/copay/issues/4423
Fixes https://github.com/bitpay/copay/issues/4427

### CODE REVIEW

File: src/js/controllers/backup.js [hunk 1]
```diff
@@ -40,11 +56,8 @@ angular.module('copayApp.controllers').controller('backupController',
       return false;
     };
 
-    $scope.backTo = function(state) {
-      if (state == 'walletHome')
-        go.walletHome();
-      else
-        go.preferences();
+    $scope.backTo = function() {
```
Comment (matiu) [inline src/js/controllers/backup.js]: The function should not be called `backTo...`  and receive not params.

Maybe just `goBack()`

File: src/js/controllers/backup.js [hunk 2]
```diff
@@ -40,11 +56,8 @@ angular.module('copayApp.controllers').controller('backupController',
...
+      go.path(prevState);
```
Comment (matiu) [inline src/js/controllers/backup.js]: add `prevState || 'walletHome' ` in case it is not defined.

File: src/js/controllers/export.js [hunk 3]
```diff
@@ -1,15 +1,29 @@
...
+
+      var config = configService.getSync();
+      var touchidAvailable = fingerprintService.isAvailable();
+      var touchidEnabled = config.touchIdFor ? config.touchIdFor[fc.credentials.walletId] : null;
+
+      if (!touchidAvailable || !touchidEnabled) return;
+
+      fingerprintService.check(fc, function(err) {
+        if (err)
+          go.path($scope.goBackToState);
```
Comment (matiu) [inline src/js/controllers/export.js]: Where is defined `$scope.goBackToState` ?

**Comment by matiu** [comment]
Good work!

I left some minor comments.

File: src/js/controllers/export.js (hunk 3 above)
Comment (JDonadio) [inline src/js/controllers/export.js]: topbar.html is used by _ng-include_ and set the $scope variable 

````
<div
  class="topbar-container"
  ng-include="'views/includes/topbar.html'"
  ng-init="titleSection='Export Wallet'; goBackToState = 'preferencesAdvanced'">
</div>
````

**Comment by JDonadio** [comment]
Updated. Thanks for comment @matiu!

**Comment by matiu** [comment]
Code ACK.

I will test it manually tomorrow for the final ACK.

**Comment by cmgustavo** [comment]
I found some issues:

* Export: asks for touchID and tries to open the system keyboard. Check if there is an autofocus or something in the input.

img_1029

* Backup: TouchID and spending password request open at the same time. 

img_1030

* Also backup: if asks only TouchID, hide background content (as when request spending password)

img_1032

**Pull request closed without merging**